dashboard/lib/services/lti.rb

```diff
@@ -4,24 +4,34 @@
 
 class Services::Lti
   def self.initialize_lti_user(id_token)
-    custom_claims = id_token[Policies::Lti::LTI_CUSTOM_CLAIMS]
     user_type = Policies::Lti.get_account_type(id_token)
+    teacher_keys = [:name, :display_name, :full_name, :family_name, :given_name]
+    student_keys = [:name, :display_name, :full_name, :given_name]
+
     user = User.new
     user.provider = User::PROVIDER_MIGRATED
     user.user_type = user_type
     if user_type == User::TYPE_TEACHER
       user.age = '21+'
-      user.name = custom_claims[:display_name] || custom_claims[:full_name] || custom_claims[:family_name] || custom_claims[:given_name]
+      user.name = get_claim_from_list(id_token, teacher_keys)
     else
-      user.name = custom_claims[:display_name] || custom_claims[:full_name] || custom_claims[:given_name]
-      user.family_name = custom_claims[:family_name]
+      user.name = get_claim_from_list(id_token, student_keys)
+      user.family_name = get_claim(id_token, :family_name)
     end
     ao = AuthenticationOption.new(
       authentication_id: Policies::Lti.generate_auth_id(id_token),
       credential_type: AuthenticationOption::LTI_V1,
-      email: custom_claims[:email],
+      email: get_claim(id_token, :email),
     )
     user.authentication_options = [ao]
     user
   end
+
+  def self.get_claim_from_list(id_token, keys_array)
+    keys_array.filter_map {|key| get_claim(id_token, key)}.compact.first
```

Comment:
Is the `compact` call necessary with `filter_map`? The docs claim the return array will only include truthy results.

Reply:
Oh, yeah you're probably right. I was initially using `.map` but then the linter suggested I use `.filter_map`.